### ISSUE TITLE
Player Pets Avoidance

### ISSUE THREAD

**Issue by CraftedRO** [TrinityCore/TrinityCore#26225]
**Description:**

> [Avoidance is a passive summoned pet ability that decreases damage taken by creature AoE Spells and effects by 90%. This passive ability is innate for all classes' pets.](https://wow.gamepedia.com/Avoidance_(passive))

**Expected behaviour:**

Summoned pet shoud have this passive ability that decreases damage taken by creature AoE Spells and effects by 90% .

**Steps to reproduce the problem:**

1. Make any class who have pet and then for example go in Pit of Saron dungeon like in my video below then attack a [Plagueborn Horror](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=36879) .
2. Notice pet does take full aoe damage from [Toxic Waste](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?spell=70274) .
3. Il bring some more examples if needed.

**Branch(es):**

3.3.5

**TC rev. hash/commit:** d0b67aecb642

TrinityCore rev. d0b67aecb642 2021-03-11 20:20:34 +0200 (3.3.5 branch) (Win64, Release, Static)

**Operating system:** 

Windows 10 x64

https://user-images.githubusercontent.com/24683355/110854113-253b3880-82bd-11eb-98f4-8cd35a6ada83.mp4

**Comment by Conscript01** [comment]
I noticed this as well, in the older release of TC... There was also no avoidance on DK Gargoyle... 
I dont think that is correct approach to add this aura to creature_template_addon, but it works :D 
but there is a problem, when unholy DK gets talent Master of Ghouls - ghoul no longer be with the name "Risen Ghoul" - instead of that it will be something like - wormeater or other random generated name and for this for some reason its not working.

**Comment by offl** [comment]
I didn't checked it but there is a chance pets(including all non-player pets) missing tons of auras

**Comment by MrSmite** [comment]
This only applies to non DK pets, not all pets:

[Culling the Herd is a pet talent located in the 3rd tier of all the pet talent trees. It increases hunter's and pet's damage dealt, each time Claw, [Bite] or [Smack] critically hit. **_This pet talent was introduced in Patch 3.3.0 to replace Avoidance, which was given as a passive treat to all non-Death Knights' pets._** ](https://wowpedia.fandom.com/wiki/Culling_the_Herd)

**Comment by CraftedRO** [comment]
01f0f83139e7 new proof at Ionar heroic , pet dies in few seconds :

https://user-images.githubusercontent.com/24683355/191957780-e544f9bb-1eee-4ae3-a9ac-d1deece6d1f4.mp4


https://user-images.githubusercontent.com/24683355/191957799-366911c4-3562-4b6b-b6a2-0f9f7a67177a.mp4

**Comment by CraftedRO** [comment]
So it's a wrong approach to add those missing avoidance auras directly in creature_template_addon ?

**Comment by MrSmite** [comment]
> So it's a wrong approach to add those missing avoidance auras directly in creature_template_addon ?

No, that should be fine. My comment was only because your screenshot showed a Deathknight pet and I wanted to make sure people knew the avoidance aura does not apply to Deathknight pets.

**Comment by Shauren** [comment]
Yes, it is a wrong approach, creature_template_addon does not apply to pets

**Comment by CraftedRO** [comment]
But dk pets shoud also have avoidance and spell_pet.cpp code that shoud gives them that at least for unholy and gargoyle doesn't seems to work from my tests.
However non unholy dk pets summoned ghoul and armies shoud also have avoidance and I think this is the spell aura for them [62137](https://www.wowhead.com/wotlk/spell=62137/avoidance).

Now if creature_template_addon is not the correct table for that (btw priest shadowfiend already have this aura there: [Mana Leech](https://www.wowhead.com/wotlk/spell=28305/mana-leech) ) what's the correct table , maybe spell_pet_auras ?

**Comment by offl** [comment]
Even summoned by creatures unexpectantly has various pet auras(e.g. scaling), e.g. imps in Kil'jaeden encounter, dunno if they can have any effect on those imps, maybe it's unique case, maybe not. Maybe when those imps were created, someone just copied template and forgot to remove something like those auras. Any way as you can see last aura was added only in Cata, previous - in WotLK

<details>

```txt
ServerToClient: SMSG_AURA_UPDATE_ALL (0x7A25) Length: 23 ConnIdx: 2 Time: 04/16/2012 21:44:12.470 Number: 288479
GUID: Full: 0xF13063FE000020D1 Type: Creature Entry: 25598 (Volatile Felfire Fiend) Low: 8401
[0] Slot: 0
[0] Spell ID: 47067 ((Serverside/Non-DB2) Stun Self)
[0] Flags: 0 (None)
[0] Level: 71
[0] Charges: 0
[0] Caster GUID: Full: 0xF13063FE000020D1 Type: Creature Entry: 25598 (Volatile Felfire Fiend) Low: 8401

ServerToClient: SMSG_AURA_UPDATE (0x3A24) Length: 16 ConnIdx: 2 Time: 04/16/2012 21:44:12.470 Number: 288510
GUID: Full: 0xF13063FE000020D1 Type: Creature Entry: 25598 (Volatile Felfire Fiend) Low: 8401
[0] Slot: 1
[0] Spell ID: 34947 ((Serverside/Non-DB2) zzOldWarlock Pet Scaling 01)
[0] Flags: 15 (EffectIndex0, EffectIndex1, EffectIndex2, NotCaster)
[0] Level: 71
[0] Charges: 0

ServerToClient: SMSG_AURA_UPDATE (0x3A24) Length: 16 ConnIdx: 2 Time: 04/16/2012 21:44:12.470 Number: 288512
GUID: Full: 0xF13063FE000020D1 Type: Creature Entry: 25598 (Volatile Felfire Fiend) Low: 8401
[0] Slot: 2
[0] Spell ID: 34956 ((Serverside/Non-DB2) zzOldWarlock Pet Scaling 02)
[0] Flags: 15 (EffectIndex0, EffectIndex1, EffectIndex2, NotCaster)
[0] Level: 71
[0] Charges: 0

ServerToClient: SMSG_AURA_UPDATE (0x3A24) Length: 16 ConnIdx: 2 Time: 04/16/2012 21:44:12.470 Number: 288513
GUID: Full: 0xF13063FE000020D1 Type: Creature Entry: 25598 (Volatile Felfire Fiend) Low: 8401
[0] Slot: 3
[0] Spell ID: 34957 ((Serverside/Non-DB2) zzOldWarlock Pet Scaling 03)
[0] Flags: 15 (EffectIndex0, EffectIndex1, EffectIndex2, NotCaster)
[0] Level: 71
[0] Charges: 0

ServerToClient: SMSG_AURA_UPDATE (0x3A24) Length: 16 ConnIdx: 2 Time: 04/16/2012 21:44:12.470 Number: 288514
GUID: Full: 0xF13063FE000020D1 Type: Creature Entry: 25598 (Volatile Felfire Fiend) Low: 8401
[0] Slot: 4
[0] Spell ID: 34958 ((Serverside/Non-DB2) zzOldWarlock Pet Scaling 04)
[0] Flags: 15 (EffectIndex0, EffectIndex1, EffectIndex2, NotCaster)
[0] Level: 71
[0] Charges: 0

ServerToClient: SMSG_AURA_UPDATE (0x3A24) Length: 16 ConnIdx: 2 Time: 04/16/2012 21:44:12.470 Number: 288515
GUID: Full: 0xF13063FE000020D1 Type: Creature Entry: 25598 (Volatile Felfire Fiend) Low: 8401
[0] Slot: 5
[0] Spell ID: 61013 ((Serverside/Non-DB2) zzOldWarlock Pet Scaling 05)
[0] Flags: 15 (EffectIndex0, EffectIndex1, EffectIndex2, NotCaster)
[0] Level: 71
[0] Charges: 0

ServerToClient: SMSG_AURA_UPDATE (0x3A24) Length: 16 ConnIdx: 2 Time: 04/16/2012 21:44:12.470 Number: 288516
GUID: Full: 0xF13063FE000020D1 Type: Creature Entry: 25598 (Volatile Felfire Fiend) Low: 8401
[0] Slot: 6
[0] Spell ID: 89953 ((Serverside/Non-DB2) zzOldWarlock Pet Scaling 06)
[0] Flags: 15 (EffectIndex0, EffectIndex1, EffectIndex2, NotCaster)
[0] Level: 71
[0] Charges: 0
